README.md

```diff
@@ -2,36 +2,104 @@
+
+The `classport.agent`
 package contains a Java agent that can be run alongside the Classport-ified JAR.
 This agent logs all the supply-chain data for classes that get loaded,
 and prints a dependency tree from these at the end. This way, the dependency tree
 will consist of only those dependencies that were actually used at runtime.
 
-Finally, the `classport.analyser` package contains two tools for statically analysing
+#### Static Analyser
+
+The `classport.analyser` package contains two tools for statically analysing
 and modifying JAR files.
 
+
 ## Usage
 
-The `classport` goal will retrieve a list of project dependencies and their
-corresponding JAR files, embed the annotation into all class files within each
-JAR, and recreate what can be seen as a dependency-only Maven local repository
-in the `classport-files` directory. The JAR files from within there can then be
-included in the class path with the `-cp` flag as per usual.
+Package all the modules. From the root of the Classport project:
+
+```console 
+mvn package
+```
+
+Use the Maven plugin to embed supply chain information into the class files of the project using the `embed` goal. From the root directory of the analysed Maven project:
+
+```console
+mvn io.github.chains-project:classport-maven-plugin:0.1.0-SNAPSHOT:embed
+```
+The `embed` goal of the Maven plugin retrieves a list of project dependencies and their corresponding JAR files, embeds the annotation into all class files within each JAR, and recreates what can be seen as a dependency-only Maven local repository in the `classport-files` directory.
+
+The JAR files from within there can then be included in the class path with the `-cp` flag as per usual.
+
+For projects that get packaged into an Uber-JAR:
+
+```console
+mvn package -Dmaven.repo.local=classport-files
+```
 
-For projects that get packaged into an Uber-JAR, add `-Dmaven.repo.local=classport-files`
-to your `mvn package` flags, as this will package the project using the amended files.
 For multi-module projects, package each project separately as dependency properties
 may differ (e.g. a direct dependency for one module is a transitive one for another).
 
-Finally, run the packaged JAR using the `-javaagent:path-to-classport-agent` flag.
+### Dynamic analysis
+Use the agent to detect the used classes:
+
+```console
+java -javaagent:<path-to-agent-jar> -jar <path-to-app-jar>
+```
+
+This command outputs the **runtime representation** of the software supply chain of the analysed project:
+- `classport-deps-list` --> flat list of dependencies
+- `classport-deps-tree` --> tree of dependnecies
+
+### Static analysis
+
+This is the command to perform static analysis:
+```console
+java -jar <path-to-analyzer-jar> -<printList|printTree|generateTestJar> <jarFile> [classes-to-be-ignored]
+```
+
+The available flags are:
+* `-printList`, for statically generating a flat list of dependencies.
+* `-printTree`, for staticaly generating a dependencies tree.
```

Comment:
Can you add example output here? It can be for the demo project that you were trying.